app/models/order.rb

```diff
@@ -12,10 +12,15 @@ class Order < ApplicationRecord
   validate :user_or_cash_or_pin
   validate :activity_not_locked
 
+  before_create :can_user_create_order?
   before_destroy -> { throw(:abort) }
 
   scope :orders_for, (->(user) { where(user: user) })
 
+  def can_user_create_order?
+    throw(:abort) unless user.can_order(activity)
```

Comment:
Should never occur in theory, but of course always good practice to check both front-end and back-end